src/main/java/de/cas/mse/exercise/SumDigPower.java

```diff
@@ -2,40 +2,39 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.Arrays;
 
 public class SumDigPower {
 
-	public List<Long> sumDigPow(long a, long b) {
+	public List<Long> sumDigPow(long lowerBound, long upperBound) {
 
 		List<Long> result = new ArrayList<Long>();
 
-		for (long i = a; i < b; i++) {
-
-			List<Long> longs = new ArrayList<Long>();
-
-			String temp = Long.toString(i);
-
-			// Split values
-			for (int j = 0; j < temp.length(); j++) {
-				longs.add(Long.valueOf(temp.substring(j, j + 1)));
-			}
-
-			// Create sum
-			long sum = 0;
-			for (int j = 1; j <= longs.size(); j++) {
-				sum += Math.pow(longs.get(j - 1), j);
-			}
-
-			// Test if sum is equal
-			if (sum == i) {
+		for (long number = lowerBound; number < upperBound; number++) {	
+			List<Long> digits = getDigits(number);
+			long sum = createSum(digits);
+			if (sum == number) {
 				result.add(sum);
 			}
-
 		}
-
-        System.out.println(result);
 		return result;
 
 	}
 
+    private long createSum(List<Long> digits) {
+        long sum = 0;
+        for (int index = 0; index < digits.size(); index++) {
+        	sum += Math.pow(digits.get(index), index+1);
```

Comment:
Hätte man evtl. noch in eine eigene Variable packen können, oder? Was meinst du?